src/main/java/software/amazon/awssdk/crt/s3/S3Client.java

```diff
@@ -24,8 +28,45 @@ public class S3Client extends CrtResource {
     public S3Client(S3ClientOptions options) throws CrtRuntimeException {
         TlsContext tlsCtx = options.getTlsContext();
         region = options.getRegion();
-        acquireNativeHandle(s3ClientNew(
-                this,
+
+        int proxyConnectionType = 0;
+        String proxyHost = null;
+        int proxyPort = 0;
+        TlsContext proxyTlsContext = null;
+        int proxyAuthorizationType = 0;
+        String proxyAuthorizationUsername = null;
+        String proxyAuthorizationPassword = null;
+
+        HttpProxyOptions proxyOptions = options.getProxyOptions();
+        if (proxyOptions != null) {
+            proxyConnectionType = proxyOptions.getConnectionType().getValue();
+            proxyHost = proxyOptions.getHost();
+            proxyPort = proxyOptions.getPort();
+            proxyTlsContext = proxyOptions.getTlsContext();
+            proxyAuthorizationType = proxyOptions.getAuthorizationType().getValue();
+            proxyAuthorizationUsername = proxyOptions.getAuthorizationUsername();
+            proxyAuthorizationPassword = proxyOptions.getAuthorizationPassword();
+        }
+
+        int environmentVariableProxyConnectionType = 0;
+        TlsConnectionOptions environmentVariableProxyTlsConnectionOptions = null;
+        int environmentVariableType = 1;
+        HttpProxyEnvironmentVariableSetting environmentVariableSetting = options.getHttpProxyEnvironmentVariableSetting();
+        if (environmentVariableSetting != null) {
+            environmentVariableProxyConnectionType = environmentVariableSetting.getConnectionType().getValue();
+            environmentVariableProxyTlsConnectionOptions = environmentVariableSetting.getTlsConnectionOptions();
+            environmentVariableType = environmentVariableSetting.getEnvironmentVariableType().getValue();
+        }
+
+        HttpMonitoringOptions monitoringOptions = options.getMonitoringOptions();
+        long monitoringThroughputThresholdInBytesPerSecond = 0;
+        int monitoringFailureIntervalInSeconds = 0;
+        if (monitoringOptions != null) {
+            monitoringThroughputThresholdInBytesPerSecond = monitoringOptions.getMinThroughputBytesPerSecond();
+            monitoringFailureIntervalInSeconds = monitoringOptions.getAllowableThroughputFailureIntervalSeconds();
+        }
```

Comment:
Trivial: Why not create something like `aws_http_proxy_options_from_java` instead?

Reply:
I found examples of both ways in our code and this felt simpler instead of trying parse each value in the C code. It is also consistent with ConnectionManager and StreamManager.